v3/src/components/slider/use-slider-animation.tsx

```diff
@@ -0,0 +1,114 @@
+import { useCallback, useEffect, useRef } from "react"
+import { ISliderModel } from "./slider-model"
+import { useInterval } from "@chakra-ui/react"
+
+interface IUseSliderAnimationProps {
+  sliderModel: ISliderModel
+  running: boolean
+  setRunning: (running: boolean) => void
+}
+
+export const useSliderAnimation = ({sliderModel, running, setRunning}: IUseSliderAnimationProps) => {
+  const tickTime = 1000/sliderModel.animationRate
+  const direction = sliderModel.animationDirection
+  const mode = sliderModel.animationMode
+  const prevDirectionRef = useRef("")
+  const maxMinHitsRef = useRef(0)
+  const tempValue = sliderModel.value
+
+  const resetSlider = useCallback((val?: number) => {
+    const dir = sliderModel.animationDirection
+    const testValue = val || sliderModel.value
+    if (dir === "lowToHigh" && testValue >= sliderModel.axis.max) sliderModel.setValue(sliderModel.axis.min)
+    if (dir === "highToLow" && testValue <= sliderModel.axis.min) sliderModel.setValue(sliderModel.axis.max)
```

Comment:
```suggestion
    if (dir === "lowToHigh" && testValue > sliderModel.axis.max) sliderModel.setValue(sliderModel.axis.min)
    if (dir === "highToLow" && testValue < sliderModel.axis.min) sliderModel.setValue(sliderModel.axis.max)
```
We don't need to set the value in the `=` cases.